### PR TITLE
#36659_send_book_requests

### PR DESCRIPTION
**Pull request by trinhducduy**

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -15,3 +15,53 @@
...
+      else {
+        var errors = getFormErrors(response.data);
+        modal.find(".modal-body > .errors").empty().prepend(errors);
+      }
+    }, "json");
+  })
+});
+
+function getFlashMessage(content, type){
+  return "<div class='alert alert-"+type+"'><a href='#' data-dismiss='alert'\
```
Comment (ThuBM) [inline app/assets/javascripts/application.js]: Add space before and after +

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -15,3 +15,53 @@
...
+    class='close'>x</a><ul><li>"+content+"</li></ul></div>";
```
Comment (ThuBM) [inline app/assets/javascripts/application.js]: Add space before and after +

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -15,3 +15,53 @@
...
+}
+
+function getFormErrors(errors) {
+  var wrapper = $("<ul class='alert alert-warning'></ul>");
+  for (i in errors) {
+    wrapper.append("<li>"+errors[i]+"</li>");
```
Comment (ThuBM) [inline app/assets/javascripts/application.js]: Add space before and after +

File: app/controllers/requests_controller.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        format.html {render :new}
+        format.json {render json: {status: "failed",
+          data: @request.errors.full_messages}}
+      end
+    end
+  end
+
+  private
+  def request_params
+    params.require(:request).permit :book_title, :author
```
Comment (ThuBM) [inline app/controllers/requests_controller.rb]: request có :author không ?

Comment (trinhducduy) [inline app/controllers/requests_controller.rb]: có anh ạ :)

**Comment by ThuBM** [comment]
ok

**Pull request merged**